sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -431,6 +455,252 @@ function main() {
+
+In this section we'll examine some example cases in which it might make sense to write a credential on your own.
+
+### Authenticating with a pre-fetched access token
+
+Our package `@azure/core-http` exports a `TokenCredential` interface which is used by the `@azure/identity` package to define a common public API for all of the Identity credentials that we offer.
+
+The `@azure/identity` library does not contain a `TokenCredential` implementation which can be constructed directly with an `AccessToken`. This is intentionally omitted as a main line scenario as access tokens expire frequently and have constrained usage. However, we understand there may be some scenarios where authenticating a service client with a pre-fetched token is necessary.
+
+In this example, `StaticTokenCredential` implements the `TokenCredential` abstraction. It takes a pre-fetched access token in its constructor as an `AccessToken` (defined on `@azure/core-http`), and simply returns that from its implementation of `getToken()`.
+
+```ts
+import { TokenCredential, AccessToken } from "@azure/core-http";
+
+class StaticTokenCredential implements TokenCredential {
+  constructor(private accessToken: AccessToken) {
+  }
+  async getToken(): Promise<AccessToken> {
+    return this.accessToken;
+  }
+}
+```
+
+Once the application has defined this credential, it can be used to authenticate Azure SDK clients. The following example shows how an application already using some other mechanism for acquiring tokens (in this case the hypothetical method `getTokenForScope()`) could use the `StaticTokenCredential` to authenticate a `SecretClient` from `@azure/keyvault-secrets`.
+
+```ts
+import { SecretClient } from "@azure/keyvault-secrets";
+
+async function main() {
+  const token = getTokenForScope("https://vault.azure.net/.default");
+
+  const credential = new StaticTokenCredential(token);
+
+  const client = new SecretClient("https://myvault.vault.azure.net/", credential);
+}
+```
+
+It should be noted when using this custom credential type, it is the responsibility of the caller to ensure that the token is valid, and contains the correct claims needed to authenticate calls from the particular service client. For instance in the above case the token must have the scope "https://vault.azure.net/.default" to authorize calls to Azure Blob Storage.
+
+### Authenticating with MSAL Directly
+
+Some applications already use the MSAL library's `ConfidentialClientApplication` or `PublicClientApplication` to authenticate portions of their application. In these cases, the application might want to use the same to authenticate Azure SDK clients, to take advantage of the token caching the MSAL client application is doing, and to prevent unnecessary authentication calls.
```

Comment:
What are `ConfidentialClientApplication` and `PublicClientApplication`? Are they concepts or interfaces or classes?